packages/events/src/event.rs

```diff
@@ -71,3 +73,92 @@ fn decode_event_attribute(attribute: &EventAttribute) -> Result<(String, String)
 fn base64_to_utf8(base64_str: &str) -> std::result::Result<String, DecodingError> {
     Ok(STANDARD.decode(base64_str)?.then(String::from_utf8)?)
 }
+
+fn contract_address(event: &Event) -> Option<AccountId> {
+    match event {
+        Event::Abci {
+            event_type: _,
+            attributes,
+        } => attributes
+            .get("_contract_address")
+            .and_then(|address| serde_json::from_value::<AccountId>(address.clone()).ok()),
+        _ => None,
+    }
+}
+
+pub fn event_is_from_contract(event: &Event, from_address: &AccountId) -> bool {
+    match contract_address(event) {
+        Some(emitting_address) => &emitting_address == from_address,
+        _ => false,
+    }
+}
```

Comment:
I prefer having these two as methods on `Event`, i.e. `event.contract_address()` and `event.is_from_contract(address)`.